app/controllers/concerns/api/v2/lookup_keys_common_controller.rb

```diff
@@ -45,18 +45,26 @@ def smart_class_parameter_id?
 
     def find_puppetclass
       @puppetclass = Puppetclass.authorized(:view_puppetclasses).find(params['puppetclass_id'])
+    rescue ActiveRecord::RecordNotFound
+      not_found({ :error => { :message => (_("Puppet class with id '%{id}' was not found") % { :id => params['puppetclass_id'] }) } })
```

Comment:
The difference is that lookup keys are nested resource and the rabl template for not_found message doesn't handle that properly. I tried to fix the template, but my attempts didn't look well.
Any idea @isratrade ? TL;DR: while processing `GET /api/puppetclasses/88888/smart_variables
` the puppet class is not found with
```
 Couldn't find Puppetclass with id=88888 (ActiveRecord::RecordNotFound)
  Rendered api/v2/errors/not_found.json.rabl within api/v2/layouts/error_layout (0.9ms)
```
but `api/v2/errors/not_found.json.rabl` returns `Resource smart_variable not found by id ''`
Any elegant solution?